src/genTypeFlow.ts

```diff
@@ -120,6 +116,10 @@ export function getLoaderTypeVal(resourceConfig: ResourceConfig, resourcePath: R
         retVal = `$PropertyType<${retVal}, '${resourceConfig.nestedPath}'>`;
     }
 
+    if (resourceConfig.isBatchResource) {
```

Comment:
small nit: can we move both cases inside `if (resourceConfig.isBatchResource) {}` and nest `if (resourceConfig.nestedPath) {}` to avoid checking for `isBatchResource` twice? Seems like having these checks be separate is what made it easy to make this mistake.